### ISSUE TITLE
`tf.reduce_*` called on the result of `tf.TensorArray.concat` of unknown rank return corrupted Tensor

### ISSUE THREAD

**Issue by mdanatg** [tensorflow/tensorflow#30686]
**System information**
- TensorFlow version (use command below): 2.0
- Python version: 3

**Describe the current behavior**

See the reproducing code. When `tf.reduce_mean` is called on the output of a `TensorArray.concat` which had a fully dynamic shape (including rank), it produces a `Tensor` of scalar shape, but whose value is a 1-element vector.

**Describe the expected behavior**

The resulting `Tensor` should have a scalar value, consistent with its static shape.

**Code to reproduce the issue**

Note that the bug does not reproduce if the unknown-shape Tensor is passed directly to `reduce_mean`, which would suggest some interaction between the two.

```
import numpy as np

def unknown_shape_constant(val):
  # This is one way to create tensors of unknown shape.
  return tf.py_function(lambda: np.array(val), (), tf.int32)

@tf.function
def reduce_mean_bug():
  arr = tf.TensorArray(tf.int32, size=1, dynamic_size=True)
  arr = arr.write(0, unknown_shape_constant([[1], [3]]))
  c = arr.concat()
  m1 = tf.reduce_mean(c)

  arr = tf.TensorArray(tf.int32, size=1, dynamic_size=True)
  arr = arr.write(0, tf.constant([[1], [3]]))
  c = arr.concat()
  m2 = tf.reduce_mean(c)

  return m1, m2

m1, m2 = reduce_mean_bug()
assert m1.shape == m2.shape  # Fails
```

**Comment by mdanatg** [comment]
@jaingaurav could you triage the issue?

**Comment by mdanatg** [comment]
@morgangiraud pointed out that the issue manifests for `tf.reduce_sum` as well, and possibly other `reduce_` ops.

**Comment by gadagashwini** [comment]
This issue is fixed in Tf-nightly ==2.2.0.dev20200312 version.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/f6bffcfcc013b2316bd29f3d6b2d2714/untitled453.ipynb). Can we close this issue. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?
